src/extraction/FStar.Extraction.ML.Syntax.fs

```diff
@@ -245,10 +245,10 @@ let ml_string_ty  = MLTY_Named ([], (["Prims"], "string"))
 let ml_unit    = with_ty ml_unit_ty (MLE_Const MLC_Unit)
 let mlp_lalloc = (["SST"], "lalloc")
 let apply_obj_repr :  mlexpr -> mlty -> mlexpr = fun x t ->
-    let obj_ns = if Options.codegen() = Some Options.FSharp
-                 then "FSharp.Compatibility.OCaml.Obj"
-                 else "Obj" in
-    let obj_repr = with_ty (MLTY_Fun(t, E_PURE, MLTY_Top)) (MLE_Name([obj_ns], "repr")) in
+    let repr_name = if Options.codegen() = Some Options.FSharp
+                    then MLE_Name([], "box")
```

Comment:
For now I've only inlined the implementation of FSharp.Compatibility.OCaml.Obj.repr.